### PR TITLE
raw overlaps to avoid unnecessary object creation

### PR DESCRIPTION
**Pull request by ghandhikus**

### CODE REVIEW

**Comment by cypherdare** [comment]
I think this needs a javadoc to explain the parameters and that it's for rectangles.

**Comment by ghandhikus** [comment]
Parameter count and names suggest that

**Comment by MobiDevelop** [comment]
I'd agree that the method needs a javadoc and/or be named something with the word rectangles in it (e.g. `overlapRectangles(...)`).

File: gdx/src/com/badlogic/gdx/math/Intersector.java [hunk 1]
```diff
@@ -887,6 +887,21 @@ public static boolean overlaps (Rectangle r1, Rectangle r2) {
 		return r1.overlaps(r2);
 	}
 
+	/**
+	 * @param r1x x of first {@link Rectangle}
+	 * @param r1y y of first {@link Rectangle}
+	 * @param r1w width of first {@link Rectangle}
+	 * @param r1h height of first {@link Rectangle}
+	 * @param r2x x of first {@link Rectangle}
```
Comment (Darkyenus) [inline gdx/src/com/badlogic/gdx/math/Intersector.java]: "second"

Nothing in the signature (except parameter names) suggest that this method deals with rectangles, I am in favor of renaming the method to something like `overlapRectangles`.

**Comment by badlogic** [comment]
Agree with the others, the method name needs to indicate that this a rectangle overlap test. Please amend the PR or send a new one.

**Pull request closed without merging**